emulsify.drush.inc

```diff
@@ -0,0 +1,299 @@
+<?php
+
+/**
+ * @file
+ * Contains Drush hooks. Inspired by Aeon drush commands.
+ */
+
+/**
+ * Implements hook_drush_command().
+ */
+function emulsify_drush_command() {
+  $items = array();
+
+  $items['emulsify'] = array(
+    'description' => 'Create an Emulsify theme.',
+    'arguments' => array(
+      'name'         => 'The name of your theme.',
+    ),
+    'options' => array(
+      'machine_name' => 'The machine-readable name of your theme. This will be auto-generated from the human-readable name if ommited.',
+      'description' => 'The description of your theme',
+      'path' => 'The destination of your theme. Defaults to "all" (themes).',
+      'slim' => 'Only copy base files',
+    ),
+    'examples' => array(
+      'drush emulsify "theme"' => 'Creates an Emulsify theme called "theme", using the default options.',
+      'drush emulsify "theme" --machine_name=theme' => 'Creates a Emulsify theme called "theme" with a specific machine name.',
+    ),
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_drush_help().
+ */
+function emulsify_drush_help($section) {
+  switch ($section) {
+    case 'drush:emulsify':
+      return dt('This command will create an Emulsify theme. See examples to get started.');
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_emulsify($name = NULL) {
+
+  // If no $name provided, abort.
+  if (!$name) {
+    drush_print(dt('Theme name missing. See help using drush emulsify --help.'));
+    return;
+  }
+
+  // Determine the theme name.
+  if (!isset($name)) {
+    $name = drush_get_option('name');
+  }
+
+  // Determine the machine name.
+  $machine_name = drush_get_option('machine_name');
+  if (!$machine_name) {
+    $machine_name = $name;
+  }
+  $machine_name = str_replace(' ', '_', strtolower($machine_name));
+  $search = array(
+    // Remove characters not valid in function names.
+    '/[^a-z0-9_]/',
+    // Functions must begin with an alpha character.
+    '/^[^a-z]+/',
+  );
+  $machine_name = preg_replace($search, '', $machine_name);
+
+  // Description of theme.
+  $description = (drush_get_option('description')) ? trim(drush_get_option('description')) : 'Theme that uses Pattern Lab v2.';
```

Comment:
We probably want to set the default to something like "Theme based on Emulsify." and if it works, it'd be even better to include a link to the info page or something. e.g. `Theme based on <a href="http://emulsify.info">Emulsify</a>.`